### ISSUE TITLE
Destroy of ibm_is_instance_group.instance_group fails with provisioning_status must be active to delete instance group

### ISSUE THREAD

**Issue by l2fprod** [IBM-Cloud/terraform-provider-ibm#2098]
### Terraform Version
- IBM Cloud Provider version 1.16.1
- Terraform v0.12.29

### Affected Resource(s)
- ibm_is_instance_group.instance_group

### Terraform Configuration Files
See https://github.com/IBM-Cloud/vpc-tutorials/tree/master/vpc-autoscale

### Expected Behavior
Destroy completes succesfully

### Actual Behavior

Destroy fails with
```
Error: Error Deleting the InstanceGroup: Load Balancer r006-c66bd695-f88b-4c7d-a4a5-ed69637d021b provisioning_status must be active to delete instance group
{
    "StatusCode": 409,
    "Headers": {
 ...,
        "X-Request-Id": [
            "9bc944a6-9297-448c-97e1-742c0d1856da"
        ],
        "X-Xss-Protection": [
            "1; mode=block"
        ]
    },
    "Result": {
        "errors": [
            {
                "code": "conflict_field",
                "message": "Load Balancer r006-c66bd695-f88b-4c7d-a4a5-ed69637d021b provisioning_status must be active to delete instance group",
                "target": {
                    "name": "load_balancer",
                    "type": "load_balancer"
                }
            }
        ],
        "trace": "9bc944a6-9297-448c-97e1-742c0d1856da"
    },
    "RawResult": null
}
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
1. `terraform destroy`

### Important Factoids
The same template used to work with IBM Cloud Provider version 1.13.1. Maybe the provider code has changed, or the backend API did.

**Comment by astha-jain** [comment]
Hi @l2fprod This is expected behavior and not an error. In such case, User must resolve the issue with Load Balancer and re-try.

**Comment by l2fprod** [comment]
@astha-jain 

If I run `terraform apply` followed by `terraform destroy`, I expect all resources to be deleted - that should be _by design_.

I should not have to wait or to retry, it is up to the provider to wait during `apply` for the Load Balancer to be in a proper state so that I can call `destroy` right after and expect it to work.

Also note that it works with version 1.13.1 of the provider but failed later. Going to retest with the latest.

**Comment by l2fprod** [comment]
I re-ran the template in https://github.com/IBM-Cloud/vpc-tutorials/tree/master/vpc-autoscale and it seems to be linked to using Load Balancer + Instance Group. If I remove everything about autoscale, the load balancer is deleted successfully but when I add the autoscale Instance Group components it starts failing. Going to try to pinpoint which resource exactly is causing the problem.

**Comment by l2fprod** [comment]
Got a better understanding of the problem.

If you provision a Load Balancer, eventually its `provisioning_status` becomes `active` and that's where terraform considers it created. But few seconds later, the load balancer status changes to `update_pending` and then back to `active` so if I run `destroy` during that time, it fails because the load balancer is in `update_pending`.

**Comment by astha-jain** [comment]
Thanks @l2fprod for helping. I will put a fix soon.

**Comment by l2fprod** [comment]
@astha-jain 

how do you plan to fix it? automatically retry until the load balancer is `active`?

**Comment by astha-jain** [comment]
yes. check and wait for LB to be active before proceeding to delete instance group.

**Comment by l2fprod** [comment]
in the meantime, I'm doing this to delete the instance_group individually, retrying a couple of times

```
n=0
until [ "$n" -ge 10 ]
do
  terraform destroy -auto-approve -target=ibm_is_instance_group.instance_group && break
  echo "failed to delete instance_group, will retry..."
  n=$((n+1))
  sleep 60
done
```

and then I delete the rest with another call to `destroy`

**Comment by syyang-in-cloud** [comment]
Hi @l2fprod and @astha-jain ... I have another issue that seems pretty similar: https://github.com/IBM-Cloud/terraform-provider-ibm/issues/1972

**Comment by anilkumarnagaraj** [comment]
Fix is available in provider version >= 1.25. 

Closing this issue.
